proxy/http/remap/unit-tests/hosts.yaml

```diff
@@ -0,0 +1,65 @@
+#/** @file
+#
+#  A brief file description
+#
+#  @section license License
+#
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# */
+#
+# good_hosts.yaml
+#
+hosts:
+  p1: &p1
+    host: p1-cache.foo.com
+    protocol:
+      - http: 80
```

Comment:
Support there are two "http" ports, say 80 and 3128. How would that be done?

Reply:
Is this a list of maps, then? It might be better to have either a list of lists, that  is a list of pairs effectively, or a map of lists. That is,
```
protocol:
  - [ http, 80 ]
  - [ http, 3128 ]
  - [ https, 443 ]
```
or
```
protocol:
  - http:
    - 80
    - 3128
  - https:
    - 443
```

Reply:
Of your two, I'd choose the latter -- implied order of an array seems like it will lead to problems.  An alternative solution would be to require the admin to list the server twice. I believe the effect of listing multiple ports would be that the server would be in the list twice, and thus get 2x the selections than a server with 1 port. That would be obvious to someone listing the server twice,  but, I think not obvious when "just" listing multiple ports.

Reply:
I'm +1 on leaving the protocol as is and just listing a server twice as @mlibbey suggests so that it is obvious to the person configuring this.